waveorder/io/reader.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import os
+import zarr
+from tifffile import TiffFile
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder):
+        """
+        loads ome-tiff files in folder into zarr or numpy arrays
+        Strategy:
+            1. read all files in folder
+            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
+            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict
+
+        :param folder: folder containing all ome-tiff files
+        :param reader: tifffile or aicsimageio
+        """
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
```

Comment:
This object should have a `shape` properties which returns the data dimensions in known order, say `TCZYX`. We can then map Micromanager ```self.frames = self.shape[0], self.channels = self.shape[1]```, etc. for convenience.

If our plan is to eventually switch to `aicsimageio`, then we should try to emulate their API as much as possible.

Reply:
This is reasonable but I want to stop short of implementing specific features that are well-enabled by aicsimageio.  (for example, per-image metadata parsing or other xml-specific operations).  Maybe this requires a rethink of the short term and long term goals of this reader.